### PR TITLE
Add image-mirrors flag to provide image mirrors

### PR DESCRIPTION
**Pull request by mirisu2**
The `defaultRegistry` flag solves the problem if images, for example, in the `deployment`, are specified as `image: user123/super-app:v1.0.0`. Then the value from `defaultRegistry` will substitute the necessary repository (possibly a private one).
I encountered an issue in a communal Kubernetes cluster when applications were deployed from Helm charts (actually, it depends on how) and the manifests specified images with a "full" path, for example, `quay.io/user123/super-app:v1.0.0`, and then I received an error in `k8s-iae` that the repository was inaccessible (because the perimeter is closed).
I solved this problem with the `image-mirrors` flag.
As an example:
```
-image-mirrors="index.docker.io=docker.priv1.local,quay.io=docker.priv1.local,registry.k8s.io=docker.priv2.local"
```
Thus, I achieve dynamic replacement of the original repository with a mirror.
Of course, it would be possible to give container permissions to map the configuration file (Docker JSON or registries.conf for CRI-O) and parse it, but this, in my opinion, would be overhead.

### CODE REVIEW

**Comment by nabokihms** [comment]
related https://github.com/deckhouse/k8s-image-availability-exporter/issues/145

File: main.go [hunk 1]
```diff
@@ -116,3 +120,15 @@ func (c *caPaths) Set(value string) error {
 	*c = append(*c, value)
 	return nil
 }
+
+func parseImageMirrors(mirrors string) map[string]string {
```
Comment (nabokihms) [inline main.go:124]: This is better to use the `flag.Var` interface.
```go
type mirrorsMap map[string]string

func (m *mirrorsMap) String() string {
  return fmt.Sprintf("%v", *m)
}

func (m *mirrorsMap) Set(value string) error {
  result := strings.Split(value, "=")
  if len(result) != 2 {
    return errors.New("invalid format")
  }
  (*m)[result[0]] = result[1]
  return nil
}
```

Then you can use it like this:
```
--name docker.io=test.io
--name ghcr.io=test.io
```

**Review by nabokihms — CHANGES_REQUESTED** [review]
Overall, it looks good; only a single improvement is suggested, and we will merge this 👍 .

One thing to say: I will translate the topic message from Russian to English, because there are contributors from other countries participating in the development of the k8s-iae. To make it clear for all of us why we decided to make this improvement.

File: main.go (hunk 1 above)
Comment (mirisu2) [reply main.go]: I totally agree

**Review by nabokihms — APPROVED** [review]

**Pull request merged**